dask/multiprocessing.py

```diff
@@ -3,10 +3,28 @@
 from toolz import curry, pipe, partial
 from .optimize import fuse, cull
 import multiprocessing
-import dill
-import pickle
 from .async import get_async # TODO: get better get
 from .context import _globals
+from sys import version
+
+if version < '3':
+    import copy_reg as copyreg
+else:
+    import copyreg
+
+def _reduce_method_descriptor(m):
+    return getattr, (m.__objclass__, m.__name__)
+
+# type(set.union) is used as a proxy to <class 'method_descriptor'>
+copyreg.pickle(type(set.union), _reduce_method_descriptor)
+
+import pickle
+import cloudpickle
+
+def _dumps(x):
+    return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+
+_loads = pickle.loads
```

Comment:
Not sure if this was intentionally meant to be `pickle` or if this should actually be `cloudpickle`. Assumed the latter and added PR ( https://github.com/dask/dask/pull/3180 ) to change it. However if I assumed incorrectly, please feel free to let me know. Would be good to understand why if this was intentional.